### PR TITLE
Update React@17.0.1

### PR DESCRIPTION
**Pull request by antonhalim**
Hi Builder Team 👋🏼 

## Description

React recently upgrade to the latest [17.0.1](https://github.com/facebook/react/blob/master/CHANGELOG.md#1701-october-22-2020), despite the major version, this seems to be a safe upgrade since it doesn't introduce any difficult breaking changes.

I'd like to install and use a single version of React if possible, and builder.io/react is one of our sub-dependency that's is on 16.x.

First time doing this here, so lmk if there is much more process here, thanks!

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -110,7 +110,7 @@
     "lint-staged": "^7.0.0",
     "prettier": "^1.4.4",
     "prompt": "^1.0.0",
-    "react-test-renderer": "^16.13.1",
+    "react-test-renderer": "^17.0.1",
```
Comment (mandx) [inline packages/react/package.json:113]: Hi @antonhalim, sorry for the late reply! Just one thing, what do you think of relaxing the version a bit, like, allowing React v16 _or_ v17? Something like:
```suggestion
    "react-test-renderer": "^16.13 || ^17.0.1",
```
My point is that we don't want to force folks to upgrade to React 17 just to get features/fixes they need after this change is made.

Comment (antonhalim) [reply packages/react/package.json:113]: Oh yeah, for sure, thanks!

**Comment by antonhalim** [comment]
@mandx thanks, whenever you're ready

**Review by mandx — DISMISSED** [review]

**Comment by antonhalim** [comment]
@mandx sounds good, didnt know it exists

**Review by mandx — APPROVED** [review]

**Pull request merged**